xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp

```diff
@@ -829,6 +825,34 @@ AVDictionary* CDVDDemuxFFmpeg::GetFFMpegOptionsFromInput()
     }
   }
 
+  // Allow ffmpeg options to be overidden by properties for both local files and http(s)
+  if (url.GetProtocol().empty() || url.IsProtocol("file") ||
+      url.IsProtocol("http") || url.IsProtocol("https"))
+  {
+    // For a local file we need the following protocol whitelist
+    if (url.GetProtocol().empty() || url.IsProtocol("file"))
+      av_dict_set(&options, "protocol_whitelist", "file,http,https,tcp,tls,crypto", 0);
+
+    static std::string ffmpegOptions[] =
+      {"user-agent", "seekable", "reconnect", "reconnect_at_eof", "reconnect_streamed",
+       "reconnect_delay_max", "icy", "icy_metadata_headers", "icy_metadata_packet"};
+
+    for (const std::string optionName : ffmpegOptions)
+    {
+      CVariant optionValue = m_pInput->GetProperty(optionName);
+      if (!optionValue.isNull())
+      {
+        CLog::Log(LOGDEBUG,
+                  "CDVDDemuxFFmpeg::GetFFMpegOptionsFromInput() overriding ffmpeg option '%s: %s'",
+                  optionName.c_str(), optionValue.asString().c_str());
+        if (optionName == "user-agent")
```

Comment:
this if else block doesn't make much sense, does it?

Reply:
It was simply to change the user-agent string similar to how it’s done with protocol options.

Reply:
It now also sets `hasUserAgent` if it's set.